src/app/store/signatures/requests.ts

```diff
@@ -0,0 +1,25 @@
+import { atom } from 'jotai';
+
+import { logger } from '@shared/logger';
+import { getRequestOrigin, StorageKey } from '@shared/utils/storage';
+import { atomWithParam } from '@app/store/utils/atom-with-params';
+import { getPayloadFromToken } from '@app/store/signatures/utils';
+
+export const requestTokenState = atomWithParam('signature?request', null);
+
+export const requestTokenPayloadState = atom(get => {
```

Comment:
Is it a problem that we keep adding new atoms but have introduced redux?

Reply:
I am replacing it with redux 👍